value/src/main/java/com/google/auto/value/processor/autovalue.vm

```diff
@@ -35,7 +35,7 @@ final class $subclass$formalTypes extends $origClass$actualTypes {
   $subclass(
 #foreach ($p in $props)
 
-      $p.type $p #if ($foreach.hasNext) , #end
+     #if ($p.nullable) ${p.nullableAnnotation} #end $p.type $p #if ($foreach.hasNext) , #end
```

Comment:
Can you check, if you haven't already, that the indentation is correct for both `@Nullable` and plain parameters? Each line should be indented four spaces further than the start of the constructor declaration.

(The need to tweak templates minutely like this is the main reason I might end up abandoning the template approach, much as I like it otherwise.)